File: python_stack/_python/OOP/BankAccount.py
class BankAccount:
    def __init__(self,int_rate,balance=0):
        self.int_rate=int_rate
        self.balance=balance

    def deposit(self, amount):
        self.balance+=amount
        return self

    def withdraw(self, amount):
        if (amount>self.balance):
            print("Insufficient funds: Charging a $5 fee")
            self.balance-=5
        else:
            self.balance-=amount
        return self

    def display_account_info(self):
        print("Balance: $"+str(self.balance))
        return self

    def yield_interest(self):
        if(self.balance>0):
            self.balance=self.balance*(self.int_rate/100)
        return self

account1 = BankAccount(1)
account2 = BankAccount(2)
account1.deposit(100).deposit(50).deposit(100).withdraw(50).yield_interest().display_account_info()
account2.deposit(300).deposit(300).withdraw(30).withdraw(40).withdraw(15).withdraw(15).yield_interest().display_account_info()

File: Java&Spring/Java OPP/BankAccount/src/BankAccount.java
import java.util.Random;

public class BankAccount {
	private String accountNum;
	private double checking;
	private double savings;
	private static int numberOfAccounts = 0;
	private static double totalBankBalance = 0;
	
	public BankAccount() {
		this.accountNum= randAccNum();
		this.checking=0d;
		this.savings=0d;
		numberOfAccounts++;
	}
	
	private String randAccNum() {
		String newAccountNumber = "";
		Random r = new Random();
		for(int i=0; i<10; i++) {
			newAccountNumber += r.nextInt(10);
		}
		return newAccountNumber;
	}

	public double getChecking() {
		return checking;
	}

	public double getSavings() {
		return savings;
	}
	
	public void deposit(double value, String accType) {
		if(value <= 0) {
			return;
		}
		if(accType == "savings") {
			savings+=value;
			totalBankBalance+=value;
		}else if (accType == "checking") {
			checking+=value;
			totalBankBalance+=value;
		}else {
			System.out.println("Invalid account type. please Choose savings or checking");
		}
	}
	
	public void withdraw(double value, String accType) {
		if(value <= 0) {
			return;
		}
		
		if(accType == "savings") {
			
			if(savings<value) {
				System.out.println("insufficient funds.");
			}
			else {
				savings-=value;
				totalBankBalance-=value;
			}
			
		}else if (accType == "checking") {
			
			if(checking<value) {
				System.out.println("insufficient funds.");
			}
			else {
				checking-=value;
				totalBankBalance-=value;
			}
			
		}else {
			
			System.out.println("Invalid account type. please Choose savings or checking");
			
		}
	}

	public static double getTotalBankBalance() {
		return totalBankBalance;
	}
	
	public String toString() {
		return "< Account no. " + accountNum + ", Total balance: $" + (savings + checking)+" >"; 
	}
	
}
